src/backend/optimizer/util/pathnode.c

```diff
@@ -4056,6 +4056,36 @@ create_hashjoin_path(PlannerInfo *root,
 	if (CdbPathLocus_IsNull(join_locus))
 		return NULL;
 
+	/*
+	 * We check if our path will be rescanned due to parametrization
+	 * from outer path after this node.
+	 * For all information see the comments in create_nestloop_path().
+	 */
+	if (!outer_path->rescannable && !bms_is_empty(required_outer))
+	{
+		MaterialPath *matouter = create_material_path(root, outer_path->parent, outer_path);
+
+		matouter->cdb_shield_child_from_rescans = true;
+
+		outer_path = (Path *) matouter;
+	}
+
+	if (!inner_path->rescannable && !bms_is_empty(required_outer))
```

Comment:
Is there any chance to be non rescannable and depends on param changing for inner part of hash join? Please describe the meaning of this code path.